crypto/rand/drbg_lib.c

```diff
@@ -885,26 +883,30 @@ static RAND_DRBG *drbg_setup(RAND_DRBG *parent)
  */
 DEFINE_RUN_ONCE_STATIC(do_rand_drbg_init)
 {
-    int ret = 1;
-
     /*
      * ensure that libcrypto is initialized, otherwise the
      * DRBG locks are not cleaned up properly
      */
     if (!OPENSSL_init_crypto(0, NULL))
         return 0;
 
-    ossl_init_thread_start(OPENSSL_INIT_THREAD_RAND);
```

Comment:
It complicates the error handling because it can't be undone.
And it is called on RAND_DRBG_get0_public and RAND_DRBG_get0_public
once there is really something to clean up.